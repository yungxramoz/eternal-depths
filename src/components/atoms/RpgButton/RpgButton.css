.btn {
    max-width: 28rem;
}