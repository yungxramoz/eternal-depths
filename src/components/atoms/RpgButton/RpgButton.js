import React from 'react'
import './RpgButton.css'

const RpgButton = ({ text, onClick, golden }) => {
  const btnClasses = ['rpgui-button btn']
  if (golden) btnClasses.push('golden')
  return (
    <button className={btnClasses.join(' ')} onClick={onClick}>
      {text}
    </button>
  )
}

export default RpgButton
