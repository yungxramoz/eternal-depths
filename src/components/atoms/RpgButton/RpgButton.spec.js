import React from 'react'
import RpgButton from './RpgButton'
import { render, screen } from '@testing-library/react'

describe('RpgButton', () => {
  it('renders correctly', () => {
    render(<RpgButton />)
    expect(screen).toMatchSnapshot('RpgButton')
  })
  it('has correct text', () => {
    render(<RpgButton text="Test" onClick={() => {}} />)
    const button = screen.getByRole('button')
    expect(button.textContent).toBe('Test')
  })

  it('calls onClick when clicked', () => {
    const mockOnClick = jest.fn()
    render(<RpgButton text="Test" onClick={mockOnClick} />)
    const button = screen.getByRole('button')
    button.click()
    expect(mockOnClick).toHaveBeenCalled()
  })

  it('has golden class when golden prop is true', () => {
    render(<RpgButton text="Test" onClick={() => {}} golden />)
    const button = screen.getByRole('button')
    expect(button.classList).toContain('golden')
  })
})
