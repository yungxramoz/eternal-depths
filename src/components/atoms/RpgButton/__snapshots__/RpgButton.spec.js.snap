// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RpgButton renders correctly: RpgButton 1`] = `
<body>
  <div>
    <button
      class="rpgui-button btn"
    />
  </div>
</body>
`;
